### PR TITLE
Fix/error long delta time

### PR DESCRIPTION
**Pull request by mariofv**

### CODE REVIEW

File: Scripts/src/Script/CameraLine.cpp [hunk 1]
```diff
@@ -163,6 +165,15 @@ float CameraLine::GetProgress(const float3& position, bool draw_lines)
...
 	}
 
+	float3 line_direction = positions[positions.size() - 1] - positions[0];
+	float3 current_position_direction = perpendicular_to_current_segment.b - positions[0];
+	dot_product = line_direction.Dot(current_position_direction);
+
+	if(dot_product < 0)
+	{
+		distance_to_start = 0.f;
+	}
```
Comment (Dartemiss) [inline Scripts/src/Script/CameraLine.cpp:175]: Dot product is your friendo

File: Scripts/src/Script/DialogueManager.cpp [hunk 2]
```diff
@@ -221,3 +228,68 @@ void DialogueManager::EnsurePlayersTouchingTheGround() const
...
+{
+	GameObject* popups_go = App->scene->GetGameObjectByName("Popups UI");
+	for (size_t i = 0; i < popups_go->children.size(); ++i)
+	{
+		pop_up_list.emplace_back(popups_go->children[i]);
+		static_cast<ComponentImage*>(popups_go->children[i]->GetComponent(Component::ComponentType::CANVAS_RENDERER))->Disable();
+	}
+}
+
+GameObject* DialogueManager::ShowPopup(const std::string popup_to_show)
```
Comment (Dartemiss) [inline Scripts/src/Script/DialogueManager.cpp:242]: const reference

File: Scripts/src/Script/DialogueManager.cpp [hunk 3]
```diff
@@ -221,3 +228,68 @@ void DialogueManager::EnsurePlayersTouchingTheGround() const
...
+	static_cast<ComponentImage*>(popup_to_disable->GetComponent(Component::ComponentType::CANVAS_RENDERER))->Disable();
+	showing_popup = false;
+}
+
+bool DialogueManager::IsPopupAvailable()
+{
+	return showing_popup;
+}
+
+bool DialogueManager::ValidPopup(const std::string popup_to_show)
```
Comment (Dartemiss) [inline Scripts/src/Script/DialogueManager.cpp:265]: const reference

File: Scripts/src/Script/Dissolver.cpp [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+// 	config.AddUInt(example->UUID, "ExampleNameforSave");
+// 	Script::Save(config);
+// }
+
+// //Use this for linking GO AND VARIABLES automatically
+// void Dissovler::Load(const Config& config)
+// {
+// 	exampleUUID = config.GetUInt("ExampleNameforSave", 0);
+// 	Script::Load(config);
+// }
```
Comment (Dartemiss) [inline Scripts/src/Script/Dissolver.cpp:107]: commented code

File: Scripts/src/Script/PlayerMovement.cpp [hunk 5]
```diff
@@ -320,9 +341,13 @@ void PlayerMovement::ComputeVerticalVelocity(float3 direction, float speed, bool
 	}
 }
 
-bool PlayerMovement::CanJump() const
+bool PlayerMovement::CanJump()
```
Comment (Dartemiss) [inline Scripts/src/Script/PlayerMovement.cpp:344]: const function?

Comment (JorxSS) [reply Scripts/src/Script/PlayerMovement.cpp:344]: cannot be const

File: Scripts/src/Script/EnemyManager.cpp [hunk 6]
```diff
@@ -131,60 +131,73 @@ void EnemyManager::SpawnEnemy(const unsigned type, const float3& spawn_position)
...
+	//			biter->owner->SetEnabled(false);
+	//			biter->owner->transform.SetGlobalMatrixTranslation(graveyard_position);
+	//		}
+	//		else
+	//		{
+	//			++current_number_of_enemies_alive;
+	//		}
+	//		enemies.emplace_back(biter);
+	//	}
+	//}
```
Comment (RickyVimon) [inline Scripts/src/Script/EnemyManager.cpp:200]: do you need this?

File: Scripts/src/Script/CameraController.cpp [hunk 7]
```diff
@@ -37,9 +37,9 @@ void CameraController::Awake()
 	ComponentScript* world_manager_component = world_manager_go->GetComponentScript("WorldManager");
 	world_manager = static_cast<WorldManager*>(world_manager_component->script);
 
-	camera_rail = static_cast<CameraLine*>(App->scene->GetGameObjectByName("Camera Rail")->GetComponentScript("CameraLine")->script);
-	camera_focus = static_cast<CameraLine*>(App->scene->GetGameObjectByName("Camera Focus")->GetComponentScript("CameraLine")->script);
-	level_path = static_cast<CameraLine*>(App->scene->GetGameObjectByName("Level Path")->GetComponentScript("CameraLine")->script);
+	//camera_rail = static_cast<CameraLine*>(App->scene->GetGameObjectByName("Camera Rail")->GetComponentScript("CameraLine")->script);
+	//camera_focus = static_cast<CameraLine*>(App->scene->GetGameObjectByName("Camera Focus")->GetComponentScript("CameraLine")->script);
+	//level_path = static_cast<CameraLine*>(App->scene->GetGameObjectByName("Level Path")->GetComponentScript("CameraLine")->script);
```
Comment (RickyVimon) [inline Scripts/src/Script/CameraController.cpp:42]: commented  code

**Comment by pikipoka** [comment]
Hola yo pinto sin salirme de la linea

**Pull request merged**